plugins/woocommerce/includes/widgets/class-wc-widget-cart.php

```diff
@@ -43,6 +43,60 @@ public function __construct() {
 		parent::__construct();
 	}
 
+	/**
+	 * Get list of Woo blocks that update the cart.
+	 *
+	 * @return array;
+	 */
+	public static function get_blocks() {
+		return array(
+			'woocommerce/handpicked-products',
+			'woocommerce/single-product',
+			'woocommerce/product-collection',
+			'woocommerce/add-to-cart-form',
+			'woocommerce/product-top-rated',
+			'woocommerce/product-tag',
+			'woocommerce/products-by-attribute',
+			'woocommerce/product-on-sale',
+			'woocommerce/product-new',
+			'woocommerce/product-category',
+			'woocommerce/product-best-sellers',
+			'woocommerce/all-products',
+		);
+	}
+
+	/**
+	 * Check if post has a block from the provided list of blocks.
+	 *
+	 * @param string[]                $blocks     Blocks to check for.
+	 * @param int|string|WP_Post|null $post       Post to check.
+	 *
+	 * @since 1.0.0
+	 * @version 1.0.0
+	 **/
+	private function has_block_from_list( $blocks, $post = null ) {
+		if ( ! $post ) {
+			$post = get_post();
+		}
+
+		if ( ! $post ) {
+			return false;
+		}
+
+		if ( ! function_exists( 'parse_blocks' ) ) {
+			return false;
+		}
+
+		$block_from_page = parse_blocks( $post->post_content );
+		$block_name      = $block_from_page[0]['blockName'];
+
+		if ( in_array( $block_name, $blocks, true ) ) {
+			return true;
+		}
+
+		return false;
+	}
```

Comment:
This logic doesn't make sense to me, why are we only checking the first block in the page? what if the first block was a generic core block?

Reply:
I think the previous logic of using strpos and mapping over the array (and breaking early) was good enough? it's the same logic that has_block uses.

Reply:
Nice catch, @senadir! You are right, `has_block` use a similar logic. I thought I was using a hacky solution!